### DISCUSSION TITLE
New Plan

### DISCUSSION THREAD

**Discussion by praveenscience** [praveenscience/Internship-LMS-FrontEnd#55] [General]
![GSSoC Logo](https://i.imgur.com/BR9Q5Pd.png)

# DevTernShip: Internship OnBoarding Platform

## Welcome & Plan

A lot of you will know that this project is a part of [GirlScript Summer of Code 2021](https://gssoc.girlscript.tech/). Getting straight to the point, if you want to know about this project, you can check out the intro: [DevTernShip Introduction](https://rb.gy/cygmlf). After a lot of thoughts, there are couple of changes that are made, one being this will be a Full Stack Project. Quick facts about this project:

* Working on an Industry Ready Project
* Learning MERN Stack
* Getting Internship at UK Companies
* Work with me? Let's Continue!

If you're interested, why not start with:

* Adding a Star :star: to this Repository
* [Subscribing to my YouTube Channel](https://rb.gy/fupw21), if you please?

## Timelines & Plan

We have divided the timelines into major three areas, where we will be focussing on a couple of things:

1. **Day 1:** Planning of Requirements & Features (Saturday, 13th March)
2. **Week 1:** Designing the Initial User Flow & Componentisation (Sunday, 14th March onwards)
3. **Week 2 to 10:** Development Activities (Saturday, 20th March onwards)

### Methodology

As this project is not your usual Open Source project, this is not going be a bug fixing or a robotic one. Please read the messages carefully and let's work as a team. We will be following Agile Methodology, specifically [Spotify Agile Model](https://rb.gy/tzvtsj) with some [Coding Standards](https://rb.gy/jmo3xp) being followed, which I'll discuss later.

#### (From Saturday, 13th March)

On the Day 1, along with the mentors, we will discuss the features and try to finalise most of them. The plan should be focussing on newbie friendly and more learning. On the grand scale of things, this project will be a simple one, but with a lot of work, split among others. So everyone's gonna learn.

Once the features are finalised, we will start the wireframe design work of User Flow and screens. This will help us in understanding the complete structure of the project. Not only that, we can componentise and split the work among everyone working on this project.

Once we have completed this planning, our Day 1 is completed. This also helps you all understand, how much important is the planning phase. I wanted to do this with you all.

#### (From Sunday, 14th March onwards)

On the Week 1, based on what we did on Day 1, we will start working on the list of pages, components like headers, footers, etc., states like logged in, features like senior professional vs. fresher using the system, identifying different types of end users for the product, etc.

We will also have the chance to split the different components, create multiple issues and let others pick and wait till we give a sign off. Once our initial design is done, week 1 gets over. We'll start with splitting of pages and components and create skeletons and we'll go. You all will understand how much these planning phases will help.

#### (From Saturday, 20th March onwards)

I know everyone will be excited for this part! After this week, we all will be getting into development. I'll give more technical details during that time. And don't forget, people who stick till the end might get a chance to work with me or do an internship under me in the UK. This section will be updated in a couple of days. For now, I have thought of a couple of things listed below:

* Full Stack with a Basic Backend
* Netlify Integration
* CI Build ES Lint Check
* Husky Integration

## Goodies

Yes, why not?

* An Internship Opportunity in the UK
* Lots of DevSwags - Stickers & Tees and Premium Subscriptions

Please react to this discussion, if you understood it. If not, please put in your valuable comments below. Let's discuss! 😁

**Comment by starrohan999** [comment]
that looks awesome @praveenscience  sir

**Comment by kanak22** [comment]
Excited to contribute and learn a lot🤭🎉

**Comment by ArchishmanSengupta** [comment]
This looks great. Can't wait to contribute more

**Comment by NidhiSingh0068** [comment]
Cool layout !! Waiting to get started soon.

**Comment by nekkantimadhu** [comment]
Excited !! 😃😀

**Comment by iamharsh1312** [comment]
Very much excited to contribute to this project

**Comment by Memphis1983** [comment]
This is amazing, Praveen! Looking forward to working with you and learning from the master blaster 😝

**Comment by abhishek96k** [comment]
Hey @praveenscience  I would like to work on this project.

**Comment by anjali1102** [comment]
@praveenscience  I am sorry, But I didn't get specifically what we gonna do in this project.

**Reply by praveenscience** [reply]
@anjali1102 We're gonna be meeting today at 10 PM IST in a [YouTube Live Stream](https://rb.gy/dfxjnd) for the Day 1 talks. We're gonna brainstorm and get requirements done. From tomorrow we'll do the design part. We'll get to know after the brainstorming is done. 😊

**Comment by aashi2912** [comment]
Very Excited to contribute and learn a lot.

**Comment by AshuKV** [comment]
Looking forward to work on this project, Much Excited!!!

**Comment by singhayushh** [comment]
I want to be a part of the backend team!

**Reply by praveenscience** [reply]
Welcome, let's make it! 😁 Ping me with the same message on Discord.

**Comment by tanishq-arya** [comment]
I would like to contribute to this project.

**Comment by pavankumaretta** [comment]
I gave shoutout intially but didnot assigned any issue to me till now please look into this.
Thanks !!

**Reply by praveenscience** [reply]
@Elizah550 I am not sure, why do you think? Because we assigned a lot of things here. Did you come to the [Live Stream](https://rb.gy/dfxjnd)?

![Final](https://user-images.githubusercontent.com/1830380/111137438-aec95f80-8576-11eb-8584-e921bafbb2ea.png)

**Reply by kanak22** [reply]
I missed the live stream 😐

**Reply by praveenscience** [reply]
@Elizah550 @kanak22 Don't worry, check it out on the link! 😊

**Comment by starrohan999** [comment]
I am interested in the front-end section @praveenscience  sir.

**Comment by pratikbuilds** [comment]
I want to work on any issue is there any issue i can work on.

**Reply by praveenscience** [reply]
Hi @ProPratik1405 We're following an Agile Scrum model not picking up issues model. You would have got an idea if you had seen the methodology section of the Readme. Also, did you try talking in the Discord Channel?

Can you please go through the videos mentioned in the Readme?
